vllm/engine/output_processor/single_step.py

```diff
@@ -90,7 +90,10 @@ def _process_sequence_group_outputs(self, seq_group: SequenceGroup,
             for parent_seq in parent_seqs
         }
         for sample in samples:
-            parent_child_dict[sample.parent_seq_id].append(sample)
+            # Guard against a KeyError which can occur if the request was
+            # aborted while the output was generated
+            if child_list := parent_child_dict.get(sample.parent_seq_id):
+                child_list.append(sample)
```

Comment:
@tjohnson31415 need to explicitly check for None because this will incorrectly exclude empty list case (would suggest doing that for the other check too, even though not technically needed there)